### PR TITLE
Fix NoMethodError undefined method 'coverage'

### PR DESCRIPTION
**Pull request by Maroo-b**
## Description
This is a fix for https://github.com/lostisland/faraday/issues/1254
As mentioned in coveralls-ruby issue: https://github.com/lemurheavy/coveralls-ruby/issues/161, using coveralls-ruby-rebon should fix the problem.

Here is a screenshot from the CI logs as a proof :) 
coveralls-logs

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ group :lint, :development do
 end
 
 group :test, :development do
-  gem 'coveralls', require: false
+  gem 'coveralls_reborn', '~> 0.21.0', require: false
```
Comment (olleolleolle) [inline Gemfile:21]: I think we can allow _any_ version of coveralls_reborn.

**Review by olleolleolle — COMMENTED** [review]
Thanks for setting this up!

File: Gemfile (hunk 1 above)
Comment (Maroo-b) [reply Gemfile]: The fix was only introduced for v "0.21.0"  but I removed the explicit version as you recommended.

**Review by iMacTia — APPROVED** [review]
Thanks for providing the solution! Together with #1256, we should be able to get coveralls back up 🎉

**Comment by Maroo-b** [comment]
@iMacTia I'm glad that this small fix was helpful. Feel free to share any issue up-for-grabs in this project and I can dedicate some free time to it.

**Comment by iMacTia** [comment]
We now get the following

```
Coverage report generated for RSpec to /home/runner/work/faraday/faraday/coverage. 1946 / 2095 LOC (92.89%) covered.
[Coveralls] Submitting to https://coveralls.io/api/v1
[Coveralls] Couldn't find a repository matching this job.
Coverage is at 92.89%.
Coverage report sent to Coveralls.
```

So it seems reports are now sent correctly to Coveralls 👍!
We can get this merged and look into that "Couldn't find a repository matching this job" separately

**Comment by iMacTia** [comment]
@Maroo-b Thanks! I'll send anything that pops in my mind, but for now I'd suggest to look at the "help wanted" and "good first issue" labels in the repo issues 🎉

**Pull request merged**